specs/fault-proof/index.md

```diff
@@ -440,16 +440,17 @@ Requests the host to prepare the L2 Output at the l2 output root `<outputroot>`.
 The L2 Output is the preimage of a
 [computed output root](../protocol/proposals.md#l2-output-commitment-construction).
 
-#### `l2-execution-witness <blocknumber>`
+#### `l2-payload-witness <payload_attributes>`
 
-Requests the host to prepare all preimages used in the execution of the block at `<blocknumber>` height.
+Requests the host to prepare all preimages used in the building of the payload specified by `<payload_attributes>`.
+`<payload_attributes>` is a hex-encoded JSON with the fields `parentBlockHash` and `payloadAttributes`.
```

Comment:
I don't think there's any reason we need to hex-encode this.  The hint is just a length prefixed string, so we can include the JSON directly for this hint and others can keep using a simple hex hash. So we'd still have literally `l2-payload-witness ` as the start of the string, followed by the JSON. The payload parsing already differs for different hint types so I don't see a problem using JSON here.

Reply:
Ah, the prefetcher in op-program does a hexdecode on the payload as part of generic parsing, but I think it's probably better to just move that and return the original string and then we can hex decode or json decode based on the hint type.

Reply:
sounds good! will update this